npm/test/clientfilemetataupload.test.ts

```diff
@@ -195,6 +201,50 @@ test("createMetadataInputBatches generates batches of the defined size", () => {
   expect(result[1]).toEqual([input4, input5])
 })
 
+test("createMetadataInputsAndFileMap returns metadata inputs and file map", () => {
+  const metadata: IFileMetadata[] = [mockMetadata1, mockMetadata2]
+  const client = new GraphqlClient("https://example.com", mockKeycloakInstance)
+  const uploadMetadata = new ClientFileMetadataUpload(client)
+
+  const { metadataInputs, matchFileMap } =
+    uploadMetadata.createMetadataInputsAndFileMap(metadata)
+
+  expect(metadataInputs).toHaveLength(2)
+  for (let i = 0; i < metadataInputs.length; i += 1) {
+    expect(metadataInputs[i].checksum).toEqual(metadata[i].checksum)
+    expect(metadataInputs[i].fileSize).toEqual(metadata[i].size)
+    expect(metadataInputs[i].originalPath).toEqual(metadata[i].path)
+    expect(metadataInputs[i].lastModified).toEqual(
+      metadata[i].lastModified.getTime()
+    )
+  }
+
+  expect(matchFileMap.size).toEqual(2)
+  for (let i = 0; i < metadataInputs.length; i += 1) {
+    expect(matchFileMap.get(i)).toEqual(metadata[i].file)
+  }
+})
+
+test("createMetadataInputsAndFileMap removes slash at beginning of metadata input's file path", async () => {
```

Comment:
I think it's worth adding a short comment about the context for this, because I don't think I'd understand the need for it if I was reading this for the first time. Something like this? `Files added by drag-and-drop include a leading slash but files added using the Browse button do not. Check that any leading slashes are be removed for consistency.`

Reply:
Ah ok. There is a comment in the [clientfilemetadata code](https://github.com/nationalarchives/tdr-transfer-frontend/blob/b0e904909616bf4d304e7d7a520c158b45917919/npm/src/clientfilemetadataupload/index.ts#L117) about it. Should I still add this comment for clarity though?

Reply:
Yeah, I think so. Comments are helpful whenever something is surprising, and I think this test is surprising.